homeassistant/components/google_assistant/smart_home.py

```diff
@@ -177,120 +178,145 @@ def entity_to_device(entity: Entity, config: Config, units: UnitSystem):
+        'thermostatTemperatureSetpointHigh':
+        celsius(attrs.get(climate.ATTR_TARGET_TEMP_HIGH), units),
+        'thermostatTemperatureSetpointLow':
+        celsius(attrs.get(climate.ATTR_TARGET_TEMP_LOW), units),
+        'thermostatHumidityAmbient':
+        attrs.get(climate.ATTR_CURRENT_HUMIDITY),
+    }
+    return {k: v for k, v in response.items() if v is not None}
 
-    if final_brightness is None:
-        final_brightness = 255 if final_state else 0
 
-    final_brightness = 100 * (final_brightness / 255)
+@QUERY_HANDLERS.register(media_player.DOMAIN)
+def query_response_media_player(
+        entity: Entity, config: Config, units: UnitSystem) -> dict:
+    """Convert a media_player entity to a QUERY response."""
+    level = entity.attributes.get(
+        media_player.ATTR_MEDIA_VOLUME_LEVEL,
+        1.0 if entity.state != STATE_OFF else 0.0)
+    # Convert 0.0-1.0 to 0-255
+    brightness = int(level * 100)
 
-    query_response = {
-        "on": final_state,
-        "online": True,
-        "brightness": int(final_brightness)
-    }
+    return {'brightness': brightness}
+
+
+@QUERY_HANDLERS.register(light.DOMAIN)
+def query_response_light(
+        entity: Entity, config: Config, units: UnitSystem) -> dict:
+    """Convert a light entity to a QUERY response."""
+    response = {}  # type: Dict[str, Any]
+
+    brightness = entity.attributes.get(light.ATTR_BRIGHTNESS)
+    if brightness is not None:
+        response['brightness'] = int(100 * (brightness / 255))
 
     supported_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
     if supported_features & \
        (light.SUPPORT_COLOR_TEMP | light.SUPPORT_RGB_COLOR):
-        query_response["color"] = {}
+        response['color'] = {}
 
         if entity.attributes.get(light.ATTR_COLOR_TEMP) is not None:
-            query_response["color"]["temperature"] = \
+            response['color']['temperature'] = \
                 int(round(color.color_temperature_mired_to_kelvin(
                     entity.attributes.get(light.ATTR_COLOR_TEMP))))
 
         if entity.attributes.get(light.ATTR_COLOR_NAME) is not None:
-            query_response["color"]["name"] = \
+            response['color']['name'] = \
                 entity.attributes.get(light.ATTR_COLOR_NAME)
 
         if entity.attributes.get(light.ATTR_RGB_COLOR) is not None:
             color_rgb = entity.attributes.get(light.ATTR_RGB_COLOR)
             if color_rgb is not None:
-                query_response["color"]["spectrumRGB"] = \
+                response['color']['spectrumRGB'] = \
                     int(color.color_rgb_to_hex(
                         color_rgb[0], color_rgb[1], color_rgb[2]), 16)
 
-    return query_response
+    return response
+
+
+def query_device(entity: Entity, config: Config, units: UnitSystem) -> dict:
+    """Take an entity and return a properly formatted device object."""
+    state = entity.state != STATE_OFF
+    defaults = {
```

Comment:
These defaults seem to be something new (based on the tests). Is this something that we forgot initially?

Reply:
The devices I originally set up had them due to the way query_device worked but it wasn't explicit and some of the new types (climate and sensors) weren't doing it.